backend/src/api/ressources/exams/controller.js

```diff
@@ -19,58 +19,141 @@ const controller = {
+          return res.status(404).send('Exercice not Found');
+        }
+        return res.status(200).json(result);
+      } catch (error) {
+        return next(error);
+      }
     } return res.status(400).send('Bad Request...');
   },
 
-  editExam(req, res) {
+  async editExam(req, res, next) {
     if (req.body) {
-      return res.status(200).json(req.body);
+      try {
+        const result = await ExamData.update(req.params.examId, req.body);
+        if (!result) {
+          return res.status(404).send('Exam not Found');
+        }
+        return res.status(200).json(result);
+      } catch (error) {
+        return next(error);
+      }
     } return res.status(400).send('Bad Request...');
   },
 
-  newQuestionOfExercice(req, res) {
+  /**
+   * Add a new question to an existing exercice
+   *
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Object} next
+   */
+  async newQuestionOfExercice(req, res, next) {
     if (req.body) {
-      return res.status(200).json(req.body);
-    } return res.status(400).send('Bad Request...');
+      try {
+        const exercice = await ExerciceData.getById(req.params.exerciceId);
+        if (!exercice) {
+          return res.status(404).send('Not found');
+        }
+        const result = await QuestionData.create(req.body, exercice);
+        return res.status(200).json(result);
+      } catch (error) {
+        return next(error);
+      }
+    }
+    return res.status(400).send('Bad Request...');
   },
 
-  editQuestionById(req, res) {
+  /**
+   * Edit a question based on the ID
+   *
+   * @param {Object} req
+   * @param {Object} res
+   * @param {Object} next
+   */
+  async editQuestionById(req, res, next) {
     if (req.body) {
-      return res.status(200).json(req.body);
-    } return res.status(400).send('Bad Request...');
+      try {
+        const result = await QuestionData.update(req.params.questionId, req.body);
+        if (!result) {
+          return res.status(404).send('Not Found');
+        }
+        return res.status(200).json(result);
+      } catch (error) {
+        return next(error);
+      }
+    }
+    return res.status(400).send('Bad Request...');
   },
 
-  deleteExamById(req, res) {
-    if (req.params.id) {
-      return res.status(200).send(req.params.id);
-    } return res.status(400).send('Bad Request...');
+  async deleteExamById(req, res, next) {
+    try {
+      res.locals.exam.remove();
+      res.status(204).send();
+    } catch (error) {
+      next(error);
+    }
   },
 
-  deleteExerciceById(req, res) {
-    if (req.params.id) {
-      return res.status(200).send(req.params.id);
+  deleteExerciceById(req, res, next) {
+    const { exercice, exam } = res.locals;
+    if (exercice) {
+      try {
+        const deletePromise = exercice.delete();
+        exam.exercices = exam.exercices.filter(id => id !== exercice.id);
+        const savePromise = exam.save();
+        Promise.all([deletePromise, savePromise]);
```

Comment:
No `await` on this Promise?

Reply:
Ouaip, t'façon je vais probablement refactor ça en utilisant $pull : https://mongoosejs.com/docs/api.html#mongoosearray_MongooseArray-pull